batch-processing/README.md

```diff
@@ -0,0 +1,379 @@
+ => [base  1/12] FROM docker.io/library/golang:1.19-bullseye                                         0.0s
+ => [internal] load build context                                                                    0.0s
+ => => transferring context: 716B                                                                    0.0s
+ => CACHED [base  2/12] RUN apt-get update &&     apt-get install -y wget build-essential pkg-confi  0.0s
+ => CACHED [base  3/12] RUN apt-get -q -y install libjpeg-dev libpng-dev libtiff-dev     libgif-dev  0.0s
+ => CACHED [base  4/12] RUN cd &&     wget https://github.com/ImageMagick/ImageMagick6/archive/6.9.  0.0s
+ => CACHED [base  5/12] WORKDIR /app                                                                 0.0s
+ => CACHED [base  6/12] COPY go.mod ./                                                               0.0s
+ => CACHED [base  7/12] COPY go.sum ./                                                               0.0s
+ => CACHED [base  8/12] RUN go mod download                                                          0.0s
+ => CACHED [base  9/12] COPY *.go ./                                                                 0.0s
+ => CACHED [base 10/12] COPY inputs /inputs                                                          0.0s
+ => CACHED [base 11/12] RUN mkdir -p /outputs                                                        0.0s
+ => CACHED [base 12/12] RUN go build -o /out                                                         0.0s
+ => exporting to image                                                                               0.0s
+ => => exporting layers                                                                              0.0s
+ => => writing image sha256:7f24030e5dc71ed23f98885644321c76b3d6709f7140391940ff06f486a33372         0.0s
+ => => naming to docker.io/library/test                                                              0.0s
+
+docker run \
+		--rm test
+=== RUN   TestGrayscaleMockError
+--- PASS: TestGrayscaleMockError (0.00s)
+=== RUN   TestGrayscaleMockCall
+--- PASS: TestGrayscaleMockCall (0.00s)
+PASS
+ok  	github.com/CodeYourFuture/immersive-go-course/batch-processing	0.011s
+```
+
+This will have run much faster, because all the steps are cached by Docker. This is called the [build cache](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache).
+
+Neat! We've now got the application and the tests running in Docker.
+
+## Specificiation
+
+So now we've got some running code, let's get started.
+
+We want a **CLI tool** that:
+
+- reads an input CSV containing URLs
+- downloads images by URL
+- processes them using ImageMagick to make them monochrome/grayscale
+- uploads the results to [Amazon AWS S3 cloud storage](https://aws.amazon.com/s3/)
+- writes an output CSV describing what it did
+
+It should run like this: `go run . --input input.csv --output output.csv`
```

Comment:
```suggestion
It should run like this: `go run . --input input.csv --output output.csv`

You probably want to modify the `main.go` file we've supplied in this directory, re-using some bits of it.
```

I'm not sure this is otherwise clear.